### PR TITLE
Assert methods updated for Test in Cat Photo Step 12

### LINKED ISSUE #51459: Test in Cat Photo Step 12 too restrictive
### Describe the Issue

The test requires an `href` attribute of exactly `https://freecatphotoapp.com`, but the instructions say to use `https://freecatphotoapp.com/`.

Note the trailing slash.

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-12

### Your code

```
      <p>See more <a href="https://freecatphotoapp.com/">cat photos</a> in our gallery.</p>
      <a href="https://freecatphotoapp.com">link to cat pictures</a>
```

### Expected behavior

The test should not care if the trailing slash is present or not, to be honest. As far as I know, the behaviour for the link would be the same.

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

We should change the test to use a regex match on the attribute to make the presence of the trailing slash optional.

Challenge file: https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dfa22d1b521be39a3de7be0.md

Bonus points to a PR that updates the tests to use specific assert methods, and remove the jQuery.

### PR DESCRIPTION
**Pull request by Riya267**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #51459

Snapshot (test passed with trailing slash)
image

Snapshot (test passed without trailing slash)
image

### CODE REVIEW

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
We should not need regex for every step, here.

We can still:

```js
const p = document.querySelector(`p > a`);
```

**Comment by Riya267** [comment]
> We should not need regex for every step, here.
> 
> We can still:
> 
> ```js
> const p = document.querySelector(`p > a`);
> ```

Sure but I would suggest to use regex in the last assert method where we are testing the entire string and the previously written test case is failing if we use this `<p><a` href="https://freecatphotoapp.com">cat photos</a>See more cat photos in our gallery.</p>`

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dfa22d1b521be39a3de7be0.md [hunk 1]
```diff
@@ -21,22 +21,23 @@ In the text of your `p` element, turn the words `cat photos` into a link by addi
 You should nest a new anchor (`a`) element within the `p` element.
 
 ```js
-assert($('p > a').length);
+const nestedAnchor = document.querySelector(`p > a`);
+assert(nestedAnchor);
```
Comment (lasjorg) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dfa22d1b521be39a3de7be0.md:25]: Using `assert.isNotNull()` might be more correct here.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dfa22d1b521be39a3de7be0.md [hunk 2]
```diff
@@ -21,22 +21,23 @@ In the text of your `p` element, turn the words `cat photos` into a link by addi
...
 ```
 
 The link's `href` value should be `https://freecatphotoapp.com`. You have either omitted the `href` value or have a typo.
 
 ```js
-const nestedAnchor = $('p > a')[0];
+const nestedAnchor = document.querySelector(`p > a`);
 assert(
-  nestedAnchor.getAttribute('href') === 'https://freecatphotoapp.com'
+  nestedAnchor.getAttribute('href') === 'https://freecatphotoapp.com' || nestedAnchor.getAttribute('href') === 'https://freecatphotoapp.com/'
```
Comment (lasjorg) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dfa22d1b521be39a3de7be0.md:33]: I noticed when I looked at this that the value returned from the `href` property on the element always has a trailing slash no matter if it was written in the code with or without it. When using `getAttribute` it changes.

So the test should be able to just check for that attribute value. I only tested this with Chrome and Firefox but I believe looking at [the specs](https://url.spec.whatwg.org/#concept-url-serializer) for it, it should be the same in all browsers.

```
document.querySelector('p > a').href === 'https://freecatphotoapp.com/'
```

That might be a bit cleaner. But both work.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dfa22d1b521be39a3de7be0.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ dashedName: step-12
 You can turn any text into a link, such as the text inside of a `p` element.
 
 ``` html
-<p>I think <a href="https://www.freecodecamp.org/">freeCodeCamp</a> is great.</p>
+<p>I think <a href="https://www.freecodecamp.org">freeCodeCamp</a> is great.</p>
 ```
 
 In the text of your `p` element, turn the words `cat photos` into a link by adding opening and closing anchor (`a`) tags around these words. Then set the `href` attribute to `https://freecatphotoapp.com/`
```
[the remark this answers is not in the document]
Comment (Riya267) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dfa22d1b521be39a3de7be0.md]: Oops i missed that ... removed slash

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dfa22d1b521be39a3de7be0.md (hunk 1 above)
Comment (Riya267) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dfa22d1b521be39a3de7be0.md]: right ... I have updated this

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dfa22d1b521be39a3de7be0.md (hunk 2 above)
Comment (Riya267) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5dfa22d1b521be39a3de7be0.md]: updated

**Review by lasjorg — APPROVED** [review]
LGTM

**Review by jdwilkin4 — APPROVED** [review]
looks good 👍

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**